crates/hir/src/source_analyzer.rs

```diff
@@ -946,14 +946,14 @@ fn resolve_hir_path_(
                 let (ty, remaining) =
                     resolver.resolve_path_in_type_ns(db.upcast(), path.mod_path())?;
                 match remaining {
-                    Some(remaining) if remaining > 1 => {
+                    Some(remaining) => {
                         if remaining + 1 == path.segments().len() {
```

Comment:
`remaining` is a confusing name, I'm guessing this bug was introduced by misinterpreting this variable as the number of remaining segments while it actually indicates the index of first unresolved segment. Could you rename it to something like `remaining_idx`?

Reply:
Sure! Thank you for suggestion!